transforms/update-react-imports.js

```diff
@@ -23,28 +40,37 @@ module.exports = function(file, api, options) {
         path.value.source.value === 'React' ||
         path.value.source.value === 'react'
       );
-    })
-    .filter(path => {
-      return (
-        path.value.specifiers.length > 0 &&
-        path.value.importKind === 'value' &&
-        path.value.specifiers.some(
-          specifier => specifier.local.name === 'React',
-        )
-      );
     });
 
-  if (reactImportPath.size() > 1) {
+  // get all namespace/default React imports
+  const reactPaths = reactImportPaths.filter(path => {
+    return (
+      path.value.specifiers.length > 0 &&
+      path.value.importKind === 'value' &&
+      path.value.specifiers.some(specifier => specifier.local.name === 'React')
+    );
+  });
+
+  if (reactPaths.size() > 1) {
     throw Error(
       'There should only be one React import. Please remove the duplicate import and try again.',
     );
   }
 
-  if (reactImportPath.size() === 0) {
+  if (reactPaths.size() === 0) {
     return null;
   }
 
-  const keepReactImport =
+  const reactPath = reactPaths.paths()[0];
+
+  const isDefaultImport = reactPath.value.specifiers.some(
+    specifier =>
+      specifier.type === 'ImportDefaultSpecifier' &&
+      specifier.local.name === 'React',
+  );
+
+  // Check to see if we should keep the React import
+  const reactImportInUse =
```

Comment:
been preferring 'predicate' style variable names for booleans lately. 
```suggestion
  const isReactImportUsed =
```